### PR TITLE
Add logIfFalse Helper Function

### PR DESCRIPTION
**Pull request by anthony-murphy**
related to #5395

if we like it. it could live next to assert in common utils

### CODE REVIEW

**Comment by curtisman** [comment]
Took me a while to get the name attest. Would 'assume' be a better name?

**Comment by anthony-murphy** [comment]
I open to other names, was just looking for something that was a synonym to assert. I don't really like assume, as semantically i don't think the mean quite fits. another idea i had was proclaim

**Comment by curtisman** [comment]
or `logIfFalse`, since this is strongly tied to logging?

File: packages/utils/telemetry-utils/src/test/utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import {attest} from "../utils";
+
+class TestLogger implements ITelemetryBaseLogger {
+    send(event: ITelemetryBaseEvent): void {
+        this.events.push(event);
+    }
+    public readonly events: ITelemetryBaseEvent[]=[];
+}
+
+describe("attest", () => {
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/test/utils.spec.ts:17]:
```suggestion
describe("logIfFalse", () => {
```

File: packages/utils/telemetry-utils/src/test/utils.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    it("attest undefined value is not undefined",()=>{
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/test/utils.spec.ts:18]:
```suggestion
    it("logIfFalseundefined value is not undefined",()=>{
```

File: packages/utils/telemetry-utils/src/test/utils.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        const logger = new TestLogger();
+        const somthing: number | undefined = undefined;
+        const val = attest(logger, somthing !== undefined, "it's undefined");
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/test/utils.spec.ts:21]:
```suggestion
        const val = logIfFalse(logger, somthing !== undefined, "it's undefined");
```

File: packages/utils/telemetry-utils/src/test/utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        assert.strictEqual(val, false);
+        assert.strictEqual(logger.events.length,1);
+    });
+    it("attest value is not undefined",()=>{
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/test/utils.spec.ts:25]:
```suggestion
    it("logIfFalse value is not undefined",()=>{
```

File: packages/utils/telemetry-utils/src/test/utils.spec.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        const logger = new TestLogger();
+        const somthing: number | undefined = 1;
+        const val = attest(logger, somthing !== undefined, "it's undefined");
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/test/utils.spec.ts:28]:
```suggestion
        const val = logIfFalse(logger, somthing !== undefined, "it's undefined");
```

File: packages/utils/telemetry-utils/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+} from "@fluidframework/common-definitions";
+
+/**
+ * Like assert, but logs only if the condition is false, rather than throwing
+ * @param logger - The logger to log with
+ * @param condition - The condition to attest too
+ * @param event - The string or event to log
+ * @returns - The outcome of the condition
+ */
+export function attest(
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/utils.ts:18]:
```suggestion
export function logIfFalse(
```

**Review by curtisman — APPROVED** [review]

**Pull request merged**